language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ServiceDeclarationNodeContext.java

```diff
@@ -73,28 +80,41 @@ public ServiceDeclarationNodeContext() {
             (2) service mod:<cursor>
             function ...
              */
-            if (this.onQualifiedNameIdentifier(context, context.getTokenAtCursor().parent())) {
+            if (QNameReferenceUtil.onQualifiedNameIdentifier(context, context.getTokenAtCursor().parent())) {
                 QualifiedNameReferenceNode qNameRef = (QualifiedNameReferenceNode) context.getTokenAtCursor().parent();
                 List<Symbol> moduleContent = QNameReferenceUtil.getModuleContent(context, qNameRef,
                         ModulePartNodeContextUtil.serviceTypeDescPredicate());
 
-                return this.getCompletionItemList(moduleContent, context);
+                completionItems.addAll(this.getCompletionItemList(moduleContent, context));
+            } else {
+                List<Symbol> typeDescs = ModulePartNodeContextUtil.serviceTypeDescContextSymbols(context);
+                completionItems.addAll(this.getCompletionItemList(typeDescs, context));
+                completionItems.add(new SnippetCompletionItem(context, Snippet.DEF_OBJECT_TYPE_DESC_SNIPPET.get()));
+                completionItems.addAll(this.getModuleCompletionItems(context));
+                if (node.onKeyword().isMissing()) {
```

Comment:
We have to check the cursor position as well right? Because this will suggest `on` for the following:
```ser<cursor>vice```
(it will be filtered out by VSCode though). WDYT?